src/containers/add-resources/AddResources.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useState } from 'react'
+
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { useModalContext } from '~/context/modal-context'
+import useSelect from '~/hooks/table/use-select'
+import useSort from '~/hooks/table/use-sort'
+import useAxios from '~/hooks/use-axios'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import { AxiosResponse } from 'axios'
+
+import AddResourceModal from '~/containers/my-resources/add-resource-modal/AddResourceModal'
+
+import { initialSort } from '~/containers/add-resources/AddResources.constants'
+import { defaultResponses, snackbarVariants } from '~/constants'
+import { ajustColumns } from '~/utils/helper-functions'
+import {
+  ErrorResponse,
+  GetResourcesParams,
+  ItemsWithCount,
+  CourseResources,
+  TableColumn,
+  RemoveColumnRules,
+  Question
+} from '~/types'
+
+type GetServiceFunction<GetResourcesParams, T> = (
```

Comment:
I think, our ServiceFunction type can replace this

Reply:
Removed created type and used ServiceFunction instead. Since AxiosResponse was used for type, removed it as well.